### PR TITLE
Fix LogRecorder usage for newer core versions

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
There is some issues on migration code from https://github.com/jenkinsci/jenkins/pull/4538 see https://github.com/jenkinsci/github-branch-source-plugin/pull/485 for full details.

This PR updates the plugin usages of affected code to make it work again and updates the baseline. Issues can be demonstrated by `mvn test -Djenkins.version=2.332.1-cb-3 -Dtest=CustomLogsTest#testCustomLogsContent -Denforcer.skip -Daccess-modifier-checker.skip` 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
   <properties>
     <revision>2.81</revision>
     <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.222.4</jenkins.version>
+    <jenkins.version>2.332</jenkins.version>
```
Comment (raul-arabaolaza) [inline pom.xml:74]: No `2.332.1` available yet

Comment (alecharp) [reply pom.xml:74]: adding a `TODO` to change it to a LTS when it's out?

File: pom.xml [hunk 2]
```diff
@@ -92,8 +92,8 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.222.x</artifactId>
-        <version>887.vae9c8ac09ff7</version>
+        <artifactId>bom-2.332.x</artifactId>
```
Comment (alecharp) [inline pom.xml:95]: LTS is not available but the BOM is?

File: pom.xml (hunk 1 above)
Comment (raul-arabaolaza) [reply pom.xml:75]: Done

**Review by jtnord — APPROVED** [review]

**Review by Dohbedoh — APPROVED** [review]

**Pull request merged**